minor(cb2-14764): add recalls object to types definition package

cb-cs opened the pull request:
## Update types definition package & test-result service

This is the types definition element of the ticket above. Once created, the recalls schema can be used to map the /recalls endpoint response body to, in addition to being used as part of a test-result payload when posting to the test-result service.

[CB2-14764](https://dvsa.atlassian.net/browse/CB2-14764)

## Changelog

- added Recalls Schema
- added as optional key on test result schema

<!--DO NOT REMOVE COMMENT. MARKS END OF CHANGES SECTION.-->

cb-cs committed:
feat(cb2-14764): created recalls schema and added to test result schema

cb-cs committed:
feat(cb2-14764): version bump

naathanbrown commented on json-definitions/v1/test-result/index.json:
```diff
@@ -278,6 +278,13 @@
     "deletionFlag": {
       "type": "boolean"
+    },
+    "recalls": {
```
This should be any of ref or null I think, because you shouldn't need to send it

naathanbrown replied:
Hmm is that true.... this could be left off as not required so ignore that, but should recalls:null be allowed? my guess is yes because of how VTM does stuff

cb-cs replied:
To your first comment, in the type it is optional:

`recalls?: RecallsSchema;`

With regards to recalls being null, it's an optional key, so I guess what would be the value in allowing them to send it with a null value? If it's null, why include it in the payload?

cb-cs replied:
Or does VTM always send it as part of their mapping, and then just assign null?

cb-cs replied:
I guess if brokering API returned 500, then they would not have anything assigned as the value in the /recalls response, so in that case could be null?

Have made recalls nullable as discussed above, I think it is a scenario that could happen.

cb-cs committed:
feat(cb2-14764): made recalls nullable as per pr comment

Daniel-Searle commented:
Can the field on the ticket be updated to reflect that hasRecall will either be boolean or null as well as have this PR link added as a comment and the spelling of manufacturer corrected on the ticket. As far as I know, recalls is optional for the manufacturers and requires them to sign up. What is the manufacturer field going to be used for?

cb-cs commented:
> Can the field on the ticket be updated to reflect that hasRecall will either be boolean or null as well as have this PR link added as a comment and the spelling of manufacturer corrected on the ticket. As far as I know, recalls is optional for the manufacturers and requires them to sign up. What is the manufacturer field going to be used for?

Updated the ticket to reflect the above, PR was also added as a comment.

The `manufacturer` is returned because it is displayed in the banners on the FE apps, and is also displayed on the certificates if present.
